src/game/piece_move.rs

```diff
@@ -0,0 +1,137 @@
+use super::piece_type::*;
+
+fn validate_moves(moves: Vec<(u8, u8)>) -> Vec<(u8, u8)> {
+    let mut valid_moves: Vec<(u8, u8)> = [].to_vec();
+    for m in moves.iter() {
+        let (m1, m2) = *m;
+        if (m1 > 1 && m1 < 8) && (m2 < 8) {
+            valid_moves.push((m1, m2))
+        }
+    }
+    return valid_moves;
+}
+
+fn get_pawn_moves(start: (u8, u8), _is_white: bool) -> Vec<(u8, u8)> {
+    let (m1, m2) = start;
+    let mut possible_moves = [].to_vec();
+    if _is_white {
+        if m1 > 0 {
+            possible_moves.push((m1 + 1, m2));
+            possible_moves.push((m1 + 1, m2 + 1));
+            if m1 == 1 {
+                possible_moves.push((m1 + 2, m2));
+            }
+            if m2 > 0 {
+                possible_moves.push((m1 + 1, m2 - 1));
+            }
+        }
+    } else {
+        if m1 < 7 {
+            possible_moves.push((m1 - 1, m2));
+            possible_moves.push((m1 - 1, m2 + 1));
+            if m1 == 6 {
+                possible_moves.push((m1 - 2, m2));
+            }
+            if m2 > 0 {
+                possible_moves.push((m1 - 1, m2 - 1));
+            }
+        }
+    }
+    return validate_moves(possible_moves);
+}
+
+fn get_moves(_is_white: bool, start: (u8, u8), piece_type: PieceType) -> Vec<(u8, u8)> {
+    match piece_type {
+        PieceType::Pawn => get_pawn_moves(start, _is_white),
```

Comment:
Compiler is complaining here because you aren't matching every permutation of `piece_type`, at least on my system:

```
error[E0004]: non-exhaustive patterns: `Empty`, `King`, `Queen` and 3 more not covered
  --> src/game/piece_move.rs:44:11
   |
44 |       match piece_type {
   |             ^^^^^^^^^^ patterns `Empty`, `King`, `Queen` and 3 more not covered
   | 
  ::: src/game/piece_type.rs:2:1
   |
2  | / pub enum PieceType {
3  | |     Empty = 0,
4  | |     King = 1,
5  | |     Queen = 2,
...  |
9  | |     Pawn = 6
10 | | }
   | |_- `game::piece_type::PieceType` defined here
   |
   = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms

error: aborting due to previous error
```